### PR TITLE
DDF-4871 Audit system subject access or failed privilege escalation

### PR DESCRIPTION
**Pull request by ahoffer**
##### ABBREVIATED REVIEW BETWEEN 2.13.X AND MASTER IS IN EFFECT
Link to 2.13.x PR: #4872
@austinsteffes @tyler30clemens 
@codice/security 

#### Ask 2 committers to review/merge the PR and tag them here.
@brjeter @blen-desta @garrettfreibott

#### How should this be tested?
Testing is at reviewer's discretrion.

#### Any background context you want to provide?
No.

#### What are the relevant tickets?
Fixes: #4871 

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [X] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: platform/security/common/src/main/java/org/codice/ddf/security/common/Security.java [hunk 1]
```diff
@@ -367,6 +369,10 @@ BundleContext getBundleContext() {
     return null;
   }
 
+  void auditFailedCodeExecutionForSystemSubject(ExecutionException e) {
```
Comment (brjeter) [inline platform/security/common/src/main/java/org/codice/ddf/security/common/Security.java]: ✏️ Should we also add `@VisibleForTesting` to this method?

File: platform/security/common/src/test/java/org/codice/ddf/security/common/SecurityTest.java [hunk 2]
```diff
@@ -156,6 +156,8 @@ public void testGetSystemSubject() throws Exception {
           assertThat(security.getSystemSubject(), not(equalTo(null)));
           return null;
         });
+
+    verify(security).auditSystemSubjectAccess();
```
Comment (brjeter) [inline platform/security/common/src/test/java/org/codice/ddf/security/common/SecurityTest.java]: 💯

**Comment by ahoffer** [comment]
build now

**Comment by brjeter** [comment]
build now

**Comment by brjeter** [comment]
If you look into the build, the build actually passed. Jenkins misreported it.

**Comment by tyler30clemens** [comment]
Hero Successful ✅ 
verified audit messages show in security logs when hit

**Pull request merged**